Source/SwiftLintFramework/Rules/Style/NumberSeparatorRule.swift

```diff
@@ -137,24 +137,27 @@ public struct NumberSeparatorRule: OptInRule, CorrectableRule, ConfigurationProv
 
         let shouldAddSeparators = clean.count >= minimumLength
 
-        for (idx, char) in reversedIfNeeded(Array(clean), reversed: !isFraction).enumerated() {
+        var idx = 0
+        for char in reversedIfNeeded(clean, reversed: !isFraction) {
+            defer { correctComponents.append(String(char)) }
+            guard char.unicodeScalars.allSatisfy(CharacterSet.decimalDigits.contains) else { continue }
+
             if idx % 3 == 0 && idx > 0 && shouldAddSeparators {
                 correctComponents.append("_")
             }
-
-            correctComponents.append(String(char))
+            idx += 1
         }
 
         let expected = reversedIfNeeded(correctComponents, reversed: !isFraction).joined()
         return (expected == number, expected)
     }
 
-    private func reversedIfNeeded<T>(_ array: [T], reversed: Bool) -> [T] {
+    private func reversedIfNeeded<T>(_ collection: T, reversed: Bool) -> [T.Element] where T: Collection {
```

Comment:
Is there a specific reason for changing this to a collection? I am by no means against it, I just wonder why you changed this when the change doesn't affect the code during execution time, or does it? Maybe the previous solution might even be faster since there's no additional array initialization step like here in the return statement, but I'm not sure about that ...

Reply:
This is purely a style change and should not affect the correctness of the program. I noticed that the call site was converting the `String` to an `Array` before passing to this function, but this shouldn't be necessary since `String` is a `Collection` and has its own implementation of `.reversed()`. Though, I would be happy to revert this change if necessary since it's not related to the fix.

Reply:
No, it's fine. I just wanted to know the rationale behind the change but it's fine for me.